core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -152,6 +154,15 @@ public static CarbonTable buildFromTableInfo(TableInfo tableInfo) {
     }
     table.hasDataMapSchema =
         null != tableInfo.getDataMapSchemaList() && tableInfo.getDataMapSchemaList().size() > 0;
+    List<DataMapSchema> dataMapSchemaList = tableInfo.getDataMapSchemaList();
+    for (DataMapSchema dataMapSchema : dataMapSchemaList) {
+      if (dataMapSchema instanceof AggregationDataMapSchema) {
+        if (!table.hasTimeSeriesDataMap) {
```

Comment:
Please don't change the carbonTable for timeseriesdatamap. Add an utility function to decide it has timeseries datamap or not